providers/implementations/storemgmt/file_store_der2obj.c

```diff
@@ -87,29 +87,18 @@ static int der2obj_decode(void *provctx, OSSL_CORE_BIO *cin, int selection,
      */
     BIO *in = ossl_bio_new_from_core_bio(provctx, cin);
     BUF_MEM *mem = NULL;
-    int err, ok;
+    int ok;
 
     if (in == NULL)
         return 0;
 
     ERR_set_mark();
     ok = (asn1_d2i_read_bio(in, &mem) >= 0);
-    /*
-     * Prune low-level ASN.1 parse errors from error queue, assuming that
-     * this is called by decoder_process() in a loop trying several formats.
-     */
-    if (!ok) {
-        err = ERR_peek_last_error();
-        if (ERR_GET_LIB(err) == ERR_LIB_ASN1
-            && (ERR_GET_REASON(err) == ASN1_R_HEADER_TOO_LONG
-                || ERR_GET_REASON(err) == ASN1_R_UNSUPPORTED_TYPE
-                || ERR_GET_REASON(err) == ERR_R_NESTED_ASN1_ERROR
-                || ERR_GET_REASON(err) == ASN1_R_NOT_ENOUGH_DATA)) {
-            ERR_pop_to_mark();
-        } else {
-            ERR_clear_last_mark();
-            goto end;
-        }
```

Comment:
I have a naive question. Didn't we hide some real errors in such a way (e.g. malformed key/unsupported format variation)?